csharp/src/Google.Protobuf/CodedInputStream.cs

```diff
@@ -306,25 +357,6 @@ internal void CheckReadEndOfStreamTag()
 
         #region Reading of tags etc
 
-        /// <summary>
-        /// Peeks at the next field tag. This is like calling <see cref="ReadTag"/>, but the
-        /// tag is not consumed. (So a subsequent call to <see cref="ReadTag"/> will return the
-        /// same value.)
-        /// </summary>
-        public uint PeekTag()

```

Comment:
we can't remove anything from public API - we need to maintain backward compatibility of the library. So this method definitely needs to stay.

Reply:
In that case the entire next tag infrastructure must stay.